tutorials-and-examples/genAI-LLM/finetuning-gemma-2b-on-l4/README.md

```diff
@@ -0,0 +1,182 @@
+* To get access to the Gemma models for deployment to GKE, you must first sign the license consent agreement then generate a Hugging Face access token.
+
+## Creating the GKE cluster with L4 nodepools
+
+Let’s start by setting a few environment variables that will be used throughout this post. You should modify these variables to meet your environment and needs. 
+
+Download the code and files used throughout the tutorial:
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/ai-on-gke
+cd ai-on-gke/tutorials-and-examples/genAI-LLM/finetuning-gemma-2b-on-l4
+```
+
+Run the following commands to set the env variables and make sure to replace `<my-project-id>`:
+
+```bash
+gcloud config set project <my-project-id>
+export PROJECT_ID=$(gcloud config get project)
+export REGION=us-central1
+export HF_TOKEN=<YOUR_HF_TOKEN>
+```
+
+> Note: You might have to rerun the export commands if for some reason you reset your shell and the variables are no longer set. This can happen for example when your Cloud Shell disconnects.
+
+Create the GKE cluster by running:
+
+```bash
+gcloud container clusters create-auto ${CLUSTER_NAME} \
+  --project=${PROJECT_ID} \
+  --region=${REGION} \
+  --release-channel=rapid \
+  --cluster-version=1.29
+```
+
+### Create a Kubernetes secret for Hugging Face credentials {:#create-secret}
+
+In {{shell_name}}, do the following:
+
+  1. Configure `kubectl` to communicate with your cluster:
+
+      ```sh
+      gcloud container clusters get-credentials ${CLUSTER_NAME} --location=${REGION}
+      ```
+
+  2. Create a Kubernetes Secret that contains the Hugging Face token:
+
+      ```sh
+      kubectl create secret generic hf-secret \
+        --from-literal=hf_api_token=${HF_TOKEN} \
+        --dry-run=client -o yaml | kubectl apply -f -
+      ```
+
+### Containerize the Code with Docker and Cloud Build
+
+1. Create a Artifact Registry Docker Repository
+
+    ```sh
+    gcloud artifacts repositories create gemma \
+        --project=${PROJECT_ID} \
+        --repository-format=docker \
+        --location=us \
+        --description="Gemma Repo"
+    ```
+
+2. Open the `cloudbuild.yaml` manifest.
+3. Edit the path of the container image
+
+```yaml
+steps:
+- name: 'gcr.io/cloud-builders/docker'
+args: [ 'build', '-t', 'us-docker.pkg.dev/${PROJECT_ID}/gemma/finetune-gemma-gpu:1.0.0', '.' ]
+images:
+- 'us-docker.pkg.dev/${PROJECT_ID}/gemma/finetune-gemma-gpu:1.0.0'
+```
+
+4. Execute the build and create inference container image.
+
+    ```sh
+    gcloud builds submit .
+    ```
+
+## Run Finetune Job on GKE
+
+1. Open the `finetune.yaml` manifest.
+2. Edit the `image` name with the container image built with Cloud Build and `NEW_MODEL` environment variable value. This `NEW_MODEL` will be the name of the model you want to save in Hugging Face.
```

Comment:
... will the name of the model "(just the name without the repo)" you want to save in Hugging Face "as a public model".

wasn't clear because the example above in the `finetune.yaml` is `google/gemma-2b`

say it'll be saved as a public model?